charmdir.go

```diff
@@ -258,7 +268,7 @@ func (zp *zipPacker) visit(path string, fi os.FileInfo, err error) error {
 	if mode&os.ModeSymlink != 0 {
 		method = zip.Store
 	}
-	if hidden || relpath == "revision" {
+	if relpath == "revision" {
```

Comment:
tbh I it might be worth leaving the "hidden || " part there, for the moment if/when we allow excluding files as well was directories. If we have a file name ".git" I think it's reasonable to exclude that too.

Then shouldIgnoreDir becomes just shouldIgnore and the logic is almost the same as before, and it's pretty trivial to add a .charmignore file later if we want to.